native/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxTreeGenerator.java

```diff
@@ -0,0 +1,600 @@
+import io.ballerina.persist.components.IfElse;
+import io.ballerina.persist.components.TypeDescriptor;
+import io.ballerina.persist.objects.Entity;
+import io.ballerina.persist.objects.FieldMetaData;
+import io.ballerina.tools.text.TextDocument;
+import io.ballerina.tools.text.TextDocuments;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Optional;
+
+
+/**
+ * Class containing methods to create and read ballerina files as syntax trees.
+ *
+ * @since 0.1.0
+ */
+public class BalSyntaxTreeGenerator {
+
+    /**
+     * method to read ballerina files.
+     */
+    public static ArrayList<Entity> getEntityRecord(Path filePath, Optional<String> module) throws IOException {
+        ArrayList<Entity> entityArray = new ArrayList<>();
+        int index = -1;
+        ArrayList<String> keys = new ArrayList<>();
+        String tableName = null;
+        int count;
+        SyntaxTree balSyntaxTree = SyntaxTree.from(TextDocuments.from(Files.readString(filePath)));
+        ModulePartNode rootNote = balSyntaxTree.rootNode();
+        NodeList<ModuleMemberDeclarationNode> nodeList = rootNote.members();
+        for (ModuleMemberDeclarationNode moduleNode : nodeList) {
+            if (moduleNode.kind() != SyntaxKind.TYPE_DEFINITION || ((TypeDefinitionNode) moduleNode)
+                    .metadata().isEmpty()) {
+                continue;
+            }
+            for (AnnotationNode annotation : ((TypeDefinitionNode) moduleNode).metadata().get().annotations()) {
+                Node annotReference = annotation.annotReference();
+                if (annotReference.kind() != SyntaxKind.QUALIFIED_NAME_REFERENCE) {
+                    continue;
+                }
+                QualifiedNameReferenceNode qualifiedNameRef = (QualifiedNameReferenceNode) annotReference;
+                if (qualifiedNameRef.identifier().text().equals("Entity") && qualifiedNameRef.modulePrefix().text()
+                        .equals(BalFileConstants.PERSIST) && annotation.annotValue().isPresent()) {
+                    index += 1;
+                    for (MappingFieldNode fieldNode : annotation.annotValue().get().fields()) {
+                        if (fieldNode.kind() != SyntaxKind.SPECIFIC_FIELD) {
+                            continue;
+                        }
+                        SpecificFieldNode specificField = (SpecificFieldNode) fieldNode;
+                        if (specificField.fieldName().kind() != SyntaxKind.IDENTIFIER_TOKEN ||
+                                specificField.valueExpr().isEmpty()) {
+                            continue;
+                        }
+                        ExpressionNode valueNode = specificField.valueExpr().get();
+                        if (((SpecificFieldNode) fieldNode).fieldName().toString().trim().equals("key")) {
```

Comment:
Shall we omit `toString()` usage?
```suggestion
                        if (((SpecificFieldNode) fieldNode).fieldName().toSourceCode().trim().equals("key")) {
```

Reply:
Updated in [843d66c](https://github.com/ballerina-platform/persist-tools/pull/15/commits/843d66c37e8c0827c9289362e00a15de702370c6)

Reply:
I omitted the usage of `toString()` for Node objects but it's not possible to avoid it in `Path` and `StringBuilder` objects